networkx/algorithms/centrality/eigenvector.py

```diff
@@ -332,6 +342,10 @@ def eigenvector_centrality_numpy(G, weight=None, max_iter=50, tol=0):
         raise nx.NetworkXPointlessConcept(
             "cannot compute centrality for the null graph"
         )
+    connectivity_func = nx.is_strongly_connected if G.is_directed() else nx.is_connected
+    if not connectivity_func(G):  # See gh-6888.
+        msg = "`eigenvector_centrality_numpy` does not give consistent results for disconnected graphs"
+        raise nx.AmbiguousSolution(msg)
```

Comment:
Another completely ignorable suggestion: just avoids another local var binding!

```suggestion
        raise nx.AmbiguousSolution(
            "`eigenvector_centrality_numpy` does not give consistent results for disconnected graphs"
        )
```